src/bidiMapper/domains/context/context.ts

```diff
@@ -161,12 +164,50 @@ export class Context {
     if (cdpEvaluateResult.exceptionDetails) {
       // Serialize exception details.
       return await this._serializeCdpExceptionDetails(
-        cdpEvaluateResult.exceptionDetails
+        cdpEvaluateResult.exceptionDetails,
+        this._evaluateStacktraceLineOffset
       );
     }
 
     return {
       result: cdpEvaluateResult.result.value!,
     };
   }
+
+  public async PROTO_scriptInvoke(
+    functionDeclaration: string,
+    args: Script.PROTO.InvokeArgument[],
+    awaitPromise: boolean
+  ): Promise<Script.PROTO.ScriptInvokeResult> {
+    // TODO sadym: add error handling for serialization/deserialization errors.
+    // https://github.com/GoogleChromeLabs/chromium-bidi/issues/57
+    const invokeAndSerializeScript = `async (...serializedArgs)=>{ return _invoke(\n${functionDeclaration}\n, serializedArgs);
+      async function _invoke(f, serializedArgs) {
+        const evaluator = (${EVALUATOR_SCRIPT});
+        const deserializedArgs = serializedArgs.map(evaluator.deserialize);
+        const resultValue =
+          ${awaitPromise ? 'await' : ''} f.apply(this, deserializedArgs);
```

Comment:
This is ever so slightly different from what I think the spec says. It does the equivalent of:
```js
var result = ...;
if (awaitPromise && result instanceof Promise) {
  result = await result;
}
return result;
```

This implementaiton is instead:
```js
var result = ...;
if (awaitPromise) {
  result = await result;
}
return result;
```

I think the difference is when a script returns a thenable that isn't a promise. Those can be passed to `Promise.resolve()` and be "resolved", but won't pass the spec's "IsPromise(evaluation status.[[Value]])" test.

A test with a thenable object would be great to see if `awaitPromise` will deal with that would be great.

Reply:
both for `script.evaluate` and `script.invoke`